File: Labs/Lab 1/lab1.py

class lab1:
    def twoSum(self, nums , target ):
        numMap = {}
        n = len(nums)

        for i in range(n):
            numMap[nums[i]] = i

        for i in range(n):
            complement = target - nums[i]
            if complement in numMap and numMap[complement] != i :
                return [i, numMap[complement]]
            
        return []
    
#test cases below
nums1 = [2, 7, 11, 15]
target1 = 9
print(lab1().twoSum(nums1, target1))

nums2 = [3, 5, 8, 12]
target2 = 7
print(lab1().twoSum(nums2, target2))

nums5 = [1000000, 2000000, 3000000, 4000000]
target5 = 5000000
print(lab1().twoSum(nums5, target5))

File: Labs/Lab 1/lab1.java
import java.util.HashMap;
import java.util.Map;

class lab1{
    public int[] twoSum(int[] nums, int target){
       Map<Integer, Integer> numsMap = new HashMap<>();
       int n = nums.length;

       for(int i = 0; i < n; i++){
            numsMap.put(nums[i], i);
       }

       for(int i = 0 ; i < n; i++){
            int complement = target - nums[i];
            if( numsMap.containsKey(complement) && numsMap.get(complement) != i){
                return new int[] {i, numsMap.get(complement)};
            }
       }

       return new int[]{};
    }
}